docs/release-procedure.md

```diff
@@ -133,7 +133,8 @@ and in case of Modin it waits either for Github releases or for tags and then ma
 a new automatic PR with version increment.
 
 You should watch for that PR and, fixing any issues if there are some, merge it
-to make new Modin release appear in `conda-forge` channel.
+to make new Modin release appear in `conda-forge` channel. For detailed instructions
+on how to ensure the PR passes CI and is merge-able, check out [this page](https://github.com/conda-forge/modin-feedstock/blob/main/HOWTO.md)!
```

Comment:
We have recently merged https://github.com/modin-project/modin/pull/4096 to replace vague links. Let's follow the strategy here too.

Also, please change the PR title to pass CI.

`DOCS-#0000: Add conda forge doc`